code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -110,7 +110,11 @@
 	parents[i] = null
 
 /obj/machinery/atmospherics/components/returnPipenetAir(datum/pipeline/reference)
-	return airs[parents.Find(reference)]
+	var/list/returned_air = new()
+	for (var/i in 1 to parents.len)
+		if (parents[i] == reference)
+			returned_air.Add(airs[i])
```

Comment:
I'd prefer if you used += and list() here rather then Add() and new(), just for consistencies sake
Worth checking to see if sourceList |= list(item, item) ends up looking like sourceList(item) or sourceList(item, item)

Reply:
Changed to use += and list(). 
I'm not sure if i understand the latter part.

Reply:
you're all good